File: trace-web/trace/src/api/ApiSrsReqd.ts
import { httpPost, httpGet, httpDelete, C_OK as _C_OK } from "./http";

export const C_OK = _C_OK;

export const add_srs_reqd = async (params: any) => {
    return await httpPost("/trace-api/srs_reqd/add_srs_reqd", params);
};

export const update_srs_reqd = async (params: any) => {
    return await httpPost("/trace-api/srs_reqd/update_srs_reqd", params);
};

export const list_srs_reqd = async (params: any) => {
    return await httpGet("/trace-api/srs_reqd/list_srs_reqd", params);
};

export const get_srs_reqd = async (params: any) => {
    return await httpGet("/trace-api/srs_reqd/get_srs_reqd", params);
};

export const delete_srs_reqd = async (params: any) => {
    return await httpDelete("/trace-api/srs_reqd/delete_srs_reqd", params);
};

File: trace-api/src/api/api_srs_reqd.py
#!/usr/bin/env python
# encoding: utf-8
# @author: ZengLei


from typing import Any
from fastapi import APIRouter
from ..obj.tobj_srs_reqd import SrsReqdForm
from ..obj.vobj_srs_reqd import SrsReqdObj
from ..obj.tobj_role import Perms
from ..obj import Resp, Page
from ..serv.serv_srs_reqd import Server
from . import try_log

router = APIRouter()
server = Server()


@router.post("/add_srs_reqd", summary="新增SRS需求细节", response_model=Resp[Any])
@try_log(perm=Perms.srs_doc_edit)
async def add_srs_reqd(form: SrsReqdForm):
    return await server.add_srs_reqd(form) 


@router.delete("/delete_srs_reqd", summary="删除SRS需求细节", response_model=Resp[Any])
@try_log(perm=Perms.srs_doc_edit)
async def delete_srs_reqd(req_id: int):
    return await server.delete_srs_reqd(req_id) 


@router.post("/update_srs_reqd", summary="更新SRS需求细节", response_model=Resp[Any])
@try_log(perm=Perms.srs_doc_edit)
async def update_srs_reqd(form: SrsReqdForm):
    return await server.update_srs_reqd(form) 


@router.get("/list_srs_reqd", summary="查询SRS需求细节列表", response_model=Resp[Page[SrsReqdObj]])
@try_log(perm=Perms.srs_doc_view)
async def list_srs_reqd(product_id: int = None, doc_id: int = None, page_index: int = 0, page_size: int = 10):
    return await server.list_srs_reqd(product_id=product_id, doc_id=doc_id, page_index=page_index, page_size=page_size)


@router.get("/get_srs_reqd", summary="查询SRS需求细节详情", response_model=Resp[SrsReqdObj])
@try_log(perm=Perms.srs_doc_view)
async def get_srs_reqd(req_id: int):
    return await server.get_srs_reqd(req_id)  
